### PR TITLE
Add work-around for spurious first byte on the UART.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: src/resources/uart_esp32.cc [hunk 1]
```diff
@@ -868,6 +868,17 @@ PRIMITIVE(create) {
...
   uart_toit_hal_set_rx_timeout(init.hal, 10);
 
+  if (GPIO_IS_VALID_GPIO(rx)) {
+    // On some chips (specifically the S3) we have received up to a byte of
+    // 1-bits when the UART is enabled.
+    // To avoid this, we wait for the time of one byte at the baud rate, and
+    // then clear the RX FIFO.
+    // The stop-bits value is too large for STOP-BITS-1_5 and STOP-BITS-2, but
+    // waiting longer is OK.
+    int wait_time = 1 + (1000 * (data_bits + stop_bits) / baud_rate);
```
Comment (kasperl) [inline src/resources/uart_esp32.cc:878]: Nit:

```suggestion
    int wait_ms = 1 + (1000 * (data_bits + stop_bits) - 1) / baud_rate;
```

**Review by kasperl — APPROVED** [review]
Looks good to me.

**Review by kasperl — COMMENTED** [review]
Indentation is still weird around line 814 and some lines forward.

**Comment by floitsch** [comment]
> Indentation is still weird around line 814 and some lines forward.

fixed.

**Pull request merged**